### PR TITLE
Bug 2090035: Adding alerts card to VM overview page

### PR DESCRIPTION
**Pull request by metalice**
Signed-off-by: Matan Schatzman <mschatzm@redhat.com>

## 📝 Description

Add alerts card to VM overview.
Alerts link will be directed to the monitoring alert.
If there is any critical alert, the accordions will be automatically opened on and show the critical alerts.

## 🎥 Demo

alerts

### CODE REVIEW

File: src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabAlerts/hook/useVMAlerts.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+const useVMAlerts: UseVMAlerts = (vm: V1VirtualMachine) => {
+  const { vmi, pods } = useVMIAndPodsForVM(vm?.metadata?.name, vm?.metadata?.namespace);
+  const vmPod = getVMIPod(vmi, pods);
+  const [query] = usePrometheusPoll({
+    endpoint: PrometheusEndpoint?.RULES,
+    namespace: vm?.metadata?.namespace,
+  });
+
+  const vmAlerts = React.useMemo(() => {
+    const data = { critical: [], warning: [], info: [] };
```
Comment (yaacov) [inline src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabAlerts/hook/useVMAlerts.tsx:31]:
```suggestion
    const data = { critical: [], warning: [], info: [] };
    const vmName = vm?.metadata?.name;
```

File: src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabAlerts/hook/useVMAlerts.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          if (res?.state === RuleStates.Firing) {
+            res?.alerts?.forEach((alert) => {
+              const podName = vmPod?.metadata?.name;
+              const alertPodName = alert?.labels?.pod;
+              const metricName = alert?.labels?.name;
+              const vmName = vm?.metadata?.name;
+              if (
+                (podName && alertPodName && alertPodName === podName) ||
+                (vmName && metricName && metricName === vmName)
+              ) {
```
Comment (yaacov) [inline src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabAlerts/hook/useVMAlerts.tsx:44]:
```suggestion
              const alertLabelPodName = alert?.labels?.pod;
              const alertLabelName = alert?.labels?.name;
              // NOTE: in 4.11 the alerts are missing a vmName label,
              //             in 4.12 we will replace the following check with alertLabelVMName === vmName
              if ( alertLabelPodName.includes(vmName) || alertLabelName === vmName ) {
```

File: src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabAlerts/hook/useVMAlerts.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    endpoint: PrometheusEndpoint?.RULES,
+    namespace: vm?.metadata?.namespace,
+  });
+
+  const vmAlerts = React.useMemo(() => {
+    const data = { critical: [], warning: [], info: [] };
+    return (
+      (query as PrometheusRulesResponse)?.data?.groups?.reduce((acc, ruleGroup) => {
+        ruleGroup?.rules?.forEach((res) => {
+          if (res?.state === RuleStates.Firing) {
```
Comment (yaacov) [inline src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabAlerts/hook/useVMAlerts.tsx:35]:
```suggestion
          if (res?.state === RuleStates.Firing && res?.labels?.kubernetes_operator_part_of === 'kubevirt') {
```

File: src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabAlerts/hook/useVMAlerts.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  const [query] = usePrometheusPoll({
+    endpoint: PrometheusEndpoint?.RULES,
+    namespace: vm?.metadata?.namespace,
+  });
+
+  const vmAlerts = React.useMemo(() => {
+    const data = { critical: [], warning: [], info: [] };
+    return (
+      (query as PrometheusRulesResponse)?.data?.groups?.reduce((acc, ruleGroup) => {
+        ruleGroup?.rules?.forEach((res) => {
```
Comment (yaacov) [inline src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabAlerts/hook/useVMAlerts.tsx:34]:
```suggestion
        ruleGroup?.rules?.forEach((rule) => {
```

Comment (metalice) [reply src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabAlerts/hook/useVMAlerts.tsx]: great

File: src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabAlerts/hook/useVMAlerts.tsx (hunk 3 above)
Comment (metalice) [reply src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabAlerts/hook/useVMAlerts.tsx]: cool

File: src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabAlerts/hook/useVMAlerts.tsx (hunk 2 above)
Comment (metalice) [reply src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabAlerts/hook/useVMAlerts.tsx]: nice

**Comment by yaacov** [comment]
@metalice needs rebase

**Pull request merged**